katas/langtonant/charles/langtonant.js

```diff
@@ -1,82 +1,107 @@
-        pointer = columnPosition + 1;
-        gridArray[rowPosition][columnPosition] = !isPainted ? '[o]' : '   ';
-        isPainted = gridArray[rowPosition][pointer] == '   ' ? false : true;
-        gridArray[rowPosition][pointer] = gridArray[rowPosition][pointer] == '   ' ? directionDown : directionUp;
-        columnPositionStep = pointer;
+    case RIGHT:
+        colIndex++;
+        newDirection = !verifyPainted(grid[rowPosition + rowIndex][colPosition + colIndex]).isPaintedNextPosition ? DOWN : UP;
         break;
-    case directionLeft:
-        pointer = columnPosition - 1;
-        gridArray[rowPosition][columnPosition] = !isPainted ? '[o]' : '   ';
-        isPainted = gridArray[rowPosition][pointer] == '   ' ? false : true;
-        gridArray[rowPosition][pointer] = gridArray[rowPosition][pointer] == '   ' ? directionUp : directionDown;
-        columnPositionStep = pointer;
+    case LEFT:
+        colIndex--;
+        newDirection = !verifyPainted(grid[rowPosition + rowIndex][colPosition + colIndex]).isPaintedNextPosition ? UP : DOWN;
         break;
     }
+    return { rowIndex, colIndex, newDirection };
 }
 
-function showGrid() {
-    for (let indexRow = 0; indexRow < arg1; indexRow++) {
-        let rowContent = '';
-        for (let indexColumn = 0; indexColumn < arg1; indexColumn++) {
-            rowContent += gridArray[indexRow][indexColumn] + '';
-        }
-        console.log(rowContent);
+export function updateValues(grid, rowPosition, colPosition, newPosition) {
+    rowPosition += newPosition.rowIndex;
+    colPosition += newPosition.colIndex;
+    grid[rowPosition][colPosition] = verifyPainted(grid[rowPosition][colPosition]);
+    grid[rowPosition][colPosition].valueBox = newPosition.newDirection;
+    return { grid, rowPosition, colPosition };
+}
+
+export function verifyPainted(nextPosition) {
+    if (nextPosition.valueBox == BLACK) {
+        nextPosition.isPaintedNextPosition = true;
+    } else {
+        nextPosition.isPaintedNextPosition = false;
     }
+    return nextPosition;
 }
 
-function main() {
-    createGrid();
-    const steps = arg2;
-    rowPositionStep = startPosition;
-    columnPositionStep = startPosition;
-    for (let index = 0; index < steps; index++) {
-        directionNextStep(rowPositionStep, columnPositionStep);
+export function changeColor(currentPosition) {
+    if (!currentPosition.isPaintedNextPosition) {
+        currentPosition.valueBox = BLACK;
+    } else {
+        currentPosition.valueBox = WHITE;
+    }
+    return currentPosition;
+}
+
+// function showGrid(grid) {
+//     let displayContent = '';
+//     for (let indexR = 0; indexR < grid.length; indexR++) {
+//         for (let indexC = 0; indexC < grid[indexR].length; indexC++) {
+//             displayContent += grid[indexR][indexC].valueBox + ' ';
+//         }
+//         displayContent += '\n';
+//     }
+//     console.log(displayContent);
+// }
+
+export function antPath(arg1, arg2) {
```

Comment:
arg1 and arg2 are not good parameter names, try to avoid names that do not describe the values they hold, maybe this should be antSteps and dimension?